File: src/java/_2095_Delete_the_Middle_Node_of_a_Linked_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        ListNode dummy = new ListNode(0, head), tail = dummy;

        while (head != null && head.next != null) {
            tail = tail.next;
            head = head.next.next;
        }

        tail.next = tail.next.next;

        return dummy.next;
    }
}

File: src/python/_2095_Delete_the_Middle_Node_of_a_Linked_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        tail = dummy

        while head and head.next:
            tail = tail.next
            head = head.next.next

        tail.next = tail.next.next

        return dummy.next
